Tests/MongoSwiftTests/MongoDatabaseTests.swift

```diff
@@ -42,6 +42,49 @@ final class MongoDatabaseTests: MongoSwiftTestCase {
                 .to(throwError(ServerError.commandError(code: 59, message: "", errorLabels: nil)))
     }
 
+    func testDropDatabase() throws {
+        let encoder = BSONEncoder()
+        let center = NotificationCenter.default
+
+        let client = try MongoClient(options: ClientOptions(eventMonitoring: true))
+        client.enableMonitoring(forEvents: .commandMonitoring)
+
+        var db = client.db(type(of: self).testDatabase)
+        var writeConcern = try WriteConcern(journal: true, w: .number(1))
+
+        let collection = db.collection("collection")
+        try collection.insertOne(["test": "blahblah"])
+
+        var expectedWriteConcerns: [WriteConcern] = [
+                                                    try WriteConcern(journal: true, w: .number(1)),
+                                                    try WriteConcern(journal: true, w: .number(1), wtimeoutMS: 123)
+                                                ]
+        let observer = center.addObserver(forName: nil, object: nil, queue: nil) { notif in
+            guard let event = notif.userInfo?["event"] as? CommandStartedEvent else {
+                return
+            }
+
```

Comment:
One more suggestion: instead of removing elements from the array, I think it might be better to instead just count the number of events seen and compare it to the length of the array at the end. That way, we can iterate over the array itself without worrying about race conditions.

this will allow us to do e.g.:
```swift
for wc in expectedWriteConcerns {
     expect(try db.drop(options: DropDatabaseOptions(writeConcern: wc))).toNot(throwError())
}
```
We can't do this as-is since we'd be removing elements from the array while iterating over it.

Reply:
good point. i'm the one who suggested the removal, we do that in the command monitoring tests. oops